### PR TITLE
Fix golang CVE-2023-29409, CVE-2023-39318, CVE-2023-39319, CVE-2023-39323, CVE-2023-39533

### PR DESCRIPTION
**Pull request by liunan-ms**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Fix CVE-2023-29409, CVE-2023-39318, CVE-2023-39319, CVE-2023-39323, CVE-2023-39533, CVE-2023-29406, CVE-2023-39325

###### Change Log  

- Bump version to 1.20.10 to address CVE-2023-29409, CVE-2023-39318, CVE-2023-39319, CVE-2023-39323, CVE-2023-39533, CVE-2023-29406, CVE-2023-39325
- Remove patch that no longer applies

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-39323
- https://nvd.nist.gov/vuln/detail/CVE-2023-29409
- https://nvd.nist.gov/vuln/detail/CVE-2023-39318
- https://nvd.nist.gov/vuln/detail/CVE-2023-39319
- https://nvd.nist.gov/vuln/detail/CVE-2023-39533
- https://nvd.nist.gov/vuln/detail/CVE-2023-29406
- https://nvd.nist.gov/vuln/detail/CVE-2023-39325

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=437720&view=results

### CODE REVIEW

File: SPECS/golang/golang.spec [hunk 1]
```diff
@@ -145,6 +143,10 @@ fi
 %{_bindir}/*
 
 %changelog
+* Mon Oct 16 2023 Nan Liu <liunan@microsoft.com> - 1.20.9-1
+- Bump version to 1.20.9 to address CVE-2023-29409, CVE-2023-39318, CVE-2023-39319, CVE-2023-39323, CVE-2023-39533
+- Remove patch that no longer applies
```
Comment (PawelWMS) [inline SPECS/golang/golang.spec:148]: It doesn't apply, because it's already part of the new version or there's a conflict when the patch is applied?

Comment (liunan-ms) [reply SPECS/golang/golang.spec:148]: The removed patch is already in the new version.

File: SPECS/golang/golang.spec [hunk 2]
```diff
@@ -13,8 +13,8 @@
 %define __find_requires %{nil}
 Summary:        Go
 Name:           golang
-Version:        1.20.7
-Release:        2%{?dist}
+Version:        1.20.9
```
Comment (mfrw) [inline SPECS/golang/golang.spec:16]: Let's move to `1.20.10` instead.
It has a few more critical CVE fixes.

**Review by mfrw — COMMENTED** [review]
Instead of `1.20.9`, lets move to `1.20.10` as it is the latest and it address a few warts with `1.20.9`

**Review by PawelWMS — APPROVED** [review]
The ptests are looking good - only known failing tests.

**Comment by liunan-ms** [comment]
Bumped up to 1.20.10, fixing some conflicts while applying the patch for CVE-2023-44487.

**Comment by PawelWMS** [comment]
> Bumped up to 1.20.10, fixing some conflicts while applying the patch for [CVE-2023-44487](https://github.com/advisories/GHSA-qppj-fm5r-hxr3).

It looks like this CVE should be fixed in version 1.20.10 and so the patch can be removed. At least that's what I'm getting from [this announcement](https://groups.google.com/g/golang-announce/c/iNNxDTCjZvo) (one of the links from your link above).

**Comment by PawelWMS** [comment]
Verified failing package tests are known issues and not related to this change.
(We need to start disabling these flaky tests...)

**Pull request merged**